test/Test/Pos/Util/ModifierSpec.hs

```diff
@@ -0,0 +1,172 @@
+                \ 'MapModifier' and keeps all the 'Just' keys"
+                (mapModifierToHashMap @Text @Text)
+        describe "insertions" $ do
+            prop
+                "the created list removes all the 'Nothing' keys from the used\
+                \ 'MapModifier' and keeps all the 'Just' keys in the association list"
+                (mapModifierToList @Text @Text)
+        describe "deletions" $ do
+            prop
+                "the created list of keys removes all those with 'Just' values and keeps\
+                \ those with 'Nothing' values"
+                (mapModifierToDeletionList @Text @Text)
+        describe "mapMaybe" $ do
+            prop
+                "the created list of keys removes all those with 'Nothing' values and\
+                \ keeps the pairs from the input list whose keys are not in the modifier"
+                (allPairsAreInMaybeList @Text @Text @Text Just)
+
+
+mapModifierFormsMonoid
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Property
+mapModifierFormsMonoid = formsMonoid
+
+insertThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertThenLookup k v mapMod =
+    let newMap = Core.insert k v mapMod
+    in maybe False (== v) $ Core.lookup (const Nothing) k newMap
+
+insertDeleteThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => HashMap k v
+    -> k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertDeleteThenLookup m k v mapMod =
+    let newMap = HM.insert k v m
+        newMapMod = Core.delete k mapMod
+    in isNothing $ Core.lookup (flip HM.lookup newMap) k newMapMod
+
+allKeysAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [k]
+    -> Core.MapModifier k v
+    -> Bool
+allKeysAreInMap ks mapMod =
+    let keys = Core.keys ks mapMod
+    in all (\k ->
+                (isJust $ Core.lookup (const Nothing) k mapMod) ||
+                (not $ Core.member k mapMod))
+       keys
+
+allValsAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [(k,v)]
+    -> Core.MapModifier k v
+    -> Bool
+allValsAreInMap kvs mapMod =
+    let vals = Core.values kvs mapMod
+    in all (\v -> (v `elem` Core.values [] mapMod || (v `elem` fmap snd kvs))) vals
+
+allPairsAreInAssocList
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [(k, v)]
+    -> Core.MapModifier k v
+    -> Bool
+allPairsAreInAssocList kvs mapMod =
+    let keysVals = Core.toList kvs mapMod
+    in all (\(k, _) ->
+                (isJust . Core.lookup (const Nothing) k) mapMod ||
+                not (Core.member k mapMod))
+       keysVals
+
+mapModifierToHashMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => Core.MapModifier k v
+    -> Bool
+mapModifierToHashMap mapMod =
+    Core.insertionsMap mapMod == (Core.insertionsMap . Core.filter isJust) mapMod
```

Comment:
This property doesn't test anything except implementation details. If you want to test that `insertionsMap` is correct, you should take empty `MapModifier`, use `insert` on it with different key value pairs, then take `insertionsMap` from it and then assert that it is same as set of key value pairs inserted into modifier.